### ISSUE TITLE
{nw} does not wait for voices/self-voicing to complete

### ISSUE THREAD

**Issue by Sophira** [renpy/renpy#4387]
I first noticed this issue using Ren'py 8.0.3 on Linux (although in order to make things easier for me I'm actually running the Windows version in a WINE prefix). **However, I have verified that this behaviour also occurs using a standard Ren'py 8.0.3 install on a Windows computer.** The issue also persists through to built distributions.

The [documentation on dialogue text tags](https://www.renpy.org/doc/html/text.html#dialogue-text-tags) states, for the `{nw}` tag:

> The no-wait tag will wait for voice and self-voicing to complete before advancing.

However, this does not appear to be the case right now. I'm getting a similar behaviour to #3155 (from over a year ago), although that issue was closed in April last year after the original poster found it to have been fixed.

It seems to be happening again, though, and this happens for both voicing and self-voicing. For example, in the following code,  Ren'py does not wait for `audio/voiceline.ogg` to finish playing before it moves onto `"A new line."`:

```python
    voice "audio/voiceline.ogg"
    "Some long piece of text.{nw}"

    "A new line."
```

If you remove the `voice` line and enable self-voicing, the same behaviour occurs - the game doesn't wait for the self-voicing to complete before it moves onto the next line.

[edit: Fixed a typo in the name "Ren'py"; no other changes.]

**Comment by Gouvernathor** [comment]
You should check on the nightly version (which follow the evolutions of the repo), and not on the 8.0.3 Release. However, 8.0.3 was released after April 2022, and I _do_ reproduce that issue on the nightly.

**Comment by Sophira** [comment]
My apologies! For some reason I didn't notice that nightly builds were available, or I would have done so. I'll remember this for next time.

Thank you for taking the time to reproduce despite this!

**Comment by renpytom** [comment]
I've solved this with a mix of a documentation change ({nw} should not pause for voice) and a behavior change ({nw} should pause for self-voicing).

**Comment by Sophira** [comment]
I just tried the nightly (after applying #4391's fix) and while this fix works as described, there's an additional wrinkle you need - if the player is playing with self-voicing enabled, you'll need to make sure you wait for even `voice`d lines, otherwise the voice gets cut off when the self-voicing for the next line plays.

**Comment by renpytom** [comment]
I believe I fixed the latter portion in 4574a50590d53367fc5f470c167278516fe57df7 .
